pkg/gui/controllers/branches_controller.go

```diff
@@ -187,7 +188,7 @@ func (self *BranchesController) setUpstream(selectedBranch *models.Branch) error
 		},
 	}
 
-	if selectedBranch.IsTrackingRemote() {
+	if selectedBranch.IsTrackingRemote() && selectedBranch.Name == checkedOutBranch.Name {
```

Comment:
Alright this is my final attempt to persuade you both @stefanhaller and @AzraelSec that we should have this option on every branch. Happy to let the majority rule but I want to make my case:

There's two kinds of anger that will be evoked depending on which UX flow we go with:
* if we show the option on all branches and a user mistakenly believes they'll be resetting the _selected_ branch onto its upstream, even if the label says 'Reset checked-out branch onto selected branch upstream' and the tooltip says 'NOTE: This will not reset the selected branch onto its upstream, it will reset the checked-out branch onto the upstream', then they'll be mad, but they'll mostly be mad at themselves.
* if we only show the option on the checked-out branch, and a user has a need to reset onto another branch's upstream but the menu item is conspicuously missing, they'll be mad _at lazygit_. As a user I get infuriated when an option is disabled or hidden for no good reason.

I would rather have 5 people accidentally reset the checked-out branch onto another branch's upstream due to not not properly reading the label/tooltip than have 1 person thinking 'how come I can't reset onto this branch's upstream when I'm allowed to do it for the checked-out branch?'

My idea with the 'upstream options' menu is that it shows a variety of menu items for an upstream regardless of whether it's checked out or not as a short-cut so that the user doesn't need to navigate to the remote branches view. Resetting to the upstream is just one of those actions I want to support.

What do we think?

Reply:
Ok, convinced. 😄

Reply:
Convinced 😄 I've dropped my last commits, as said before. The PR should be ready again.

Reply:
Awesome :)